build/src/lib.rs

```diff
@@ -472,7 +472,7 @@ impl BinaryBuilder {
     }
 
     /// Validates the binaries by checking if they can be loaded by nlprule.
-    pub fn validate(self) -> Result<Self> {
+    pub fn validate(self) -> Result<()> {
```

Comment:
Let's not consume `self` here then so that e. g. `.outputs()` can still be called afterwards.

```suggestion
    pub fn validate(&self) -> Result<()> {
```